lib/matplotlib/tests/test_backend_qt4.py

```diff
@@ -7,32 +7,59 @@
 
 import pytest
 
-try:
-    import PyQt4
-except (ImportError, RuntimeError):  # RuntimeError if PyQt5 already imported.
+
+@pytest.fixture(autouse=True)
+def qt4_module():
+    try:
+        import PyQt4
+    # RuntimeError if PyQt5 already imported.
+    except (ImportError, RuntimeError):
+        try:
+            import PySide
+        except ImportError:
+            pytest.skip("Failed to import a Qt4 binding.")
+
+    qt_compat = pytest.importorskip('matplotlib.backends.qt_compat')
+    QtCore = qt_compat.QtCore
+
     try:
-        import PySide
-    except ImportError:
-        pytestmark = pytest.mark.skip("Failed to import a Qt4 binding.")
+        py_qt_ver = int(QtCore.PYQT_VERSION_STR.split('.')[0])
+    except AttributeError:
+        py_qt_ver = QtCore.__version_info__[0]
+
+    if py_qt_ver != 4:
+        pytest.skip(reason='Qt4 is not available')
+
+    from matplotlib.backends.backend_qt4 import (
+        MODIFIER_KEYS, SUPER, ALT, CTRL, SHIFT)  # noqa
+
+    mods = {}
+    keys = {}
+    for name, index in zip(['Alt', 'Control', 'Shift', 'Super'],
+                           [ALT, CTRL, SHIFT, SUPER]):
+        _, mod, key = MODIFIER_KEYS[index]
+        mods[name + 'Modifier'] = mod
+        keys[name + 'Key'] = key
 
-qt_compat = pytest.importorskip('matplotlib.backends.qt_compat')
-QtCore = qt_compat.QtCore
+    return QtCore, mods, keys
 
-from matplotlib.backends.backend_qt4 import (
-    MODIFIER_KEYS, SUPER, ALT, CTRL, SHIFT)  # noqa
 
-_, ControlModifier, ControlKey = MODIFIER_KEYS[CTRL]
-_, AltModifier, AltKey = MODIFIER_KEYS[ALT]
-_, SuperModifier, SuperKey = MODIFIER_KEYS[SUPER]
-_, ShiftModifier, ShiftKey = MODIFIER_KEYS[SHIFT]
+@pytest.fixture
+def qt_key(request):
```

Comment:
Can these just take qt4_module as arg?  (and so on for the following ones)

Reply:
They can; I did not do that yet as I was trying to figure out if we can combine these two files somehow.